daemon/containerd/service.go

```diff
@@ -195,10 +195,11 @@ func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID st
 	}
 
 	chainIDs := identity.ChainIDs(img.RootFS.DiffIDs)
-	virtualSize, err := computeVirtualSize(chainIDs, snapshotSizeFn)
+	snapShotSize, err := computeSnapshotSize(chainIDs, snapshotSizeFn)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	return usage.Size, usage.Size + virtualSize, nil
+	// TODO(thaJeztah): include content-store size for the image (similar to "GET /images/json")
+	return usage.Size, usage.Size + snapShotSize, nil
```

Comment:
wasn't 100% sure here, but given that a container is "image size" + "writable layer size", I _think_ we're missing a size here in the total (if I'm wrong, I'll remove this TODO 😂)

Reply:
I think it makes sense NOT to include the content-store size. Once created, snapshot exists independently from the content. We could even delete the content and the snapshot will not be affected.

Reply:
Hmm.. right, yeah. Thinking how the parallel is with pre-snapshottee; I guess in that case we have _mounts_ for the container, and I _think_ we also allow the image to be deleted if there's still a container using it (in which case those layers basically get "re-parented" and now are only used by the container, so will be garbage-collected once the container is gone.


/cc @rumpl @tianon @tonistiigi any thoughts what makes most sense to include?

Reply:
As I noted in the meeting today, I'd personally be surprised to find that the `docker image ls` listed size doesn't include the layers still in the content store as well as the extracted content in the snapshotter.  Just because we _can_ remove the image from the content store when it's extracted doesn't mean we would or even should, and IMO the default top-level size field should absolutely reflect all parts of "this image taking up space" including any content and all snapshotters (ideally).

It'd be great if we could dial into that somehow (maybe a separate struct in the API response that includes the individual details of the content store and each snapshotter/snapshot size?), but I don't think that's critical (especially given we can mostly query that from containerd itself outside of Docker if absolutely necessary).

Reply:
This affects `docker container ls` (size of the container), not `docker image ls` :).

Reply:
- It's somewhat orthogonal to this PR (just something I noticed while working on it), so I created a tracking ticket; https://github.com/moby/moby/issues/45351

Reply:
Ahhhh, the "xxxB (virtual yyyB)" value, hmm - so that should probably be "xxx" is the size of the RW layer and "yyy" is that _plus_ whatever value `docker image ls` shows, right?  (it'd be weird if `docker image ls` tells me 20MiB, then a container tells me only 10MiB or something)

Reply:
Yup! That's the one 🙈